redash/handlers/dashboards.py

```diff
@@ -25,8 +25,11 @@ def get(self):
 class DashboardListResource(BaseResource):
     @require_permission('list_dashboards')
     def get(self):
-        dashboards = [d.to_dict() for d in models.Dashboard.all(self.current_org, self.current_user.groups, self.current_user)]
-        return dashboards
+        results = models.Dashboard.all(self.current_org, self.current_user.groups, self.current_user)
+        page = request.args.get('page', 1, type=int)
+        page_size = request.args.get('page_size', 25, type=int)
+        dashboards = models.Dashboard.all(self.current_org, self.current_user.groups, self.current_user)
+        return paginate(results, page, page_size, lambda q: q.to_dict())
```

Comment:
If you return paginated results, then you can't implement tag filtering or search in the client side. Otherwise it will only apply to the current page you're on.